src/utility/STM32Cube_FW/shci_tl.c

```diff
@@ -323,11 +325,12 @@ static void OutputEvtTrace(TL_EvtPacket_t *phcievtbuffer)
 /* Weak implementation ----------------------------------------------------------------*/
 __WEAK void shci_cmd_resp_wait(uint32_t timeout)
 {
-  (void)timeout;
-
   CmdRspStatusFlag = SHCI_TL_CMD_RESP_WAIT;
-  while (CmdRspStatusFlag != SHCI_TL_CMD_RESP_RELEASE);
-
+  for (unsigned long start = millis(); (millis() - start) < timeout;) {
+  if (CmdRspStatusFlag == SHCI_TL_CMD_RESP_RELEASE) {
+     return;
+    }
+  }
```

Comment:
```suggestion
  for (unsigned long start = millis(); (millis() - start) < timeout;) {
  if (CmdRspStatusFlag == SHCI_TL_CMD_RESP_RELEASE) {
     break;
    }
  }
```

Note that return at the end of a void return function is not needed. And probably not MISRA compliant

Reply:
ok